payment-method-messaging/src/test/java/com/stripe/android/paymentmethodmessaging/element/analytics/FakeDurationProvider.kt

```diff
@@ -0,0 +1,30 @@
+package com.stripe.android.paymentmethodmessaging.element.analytics
+
+import com.stripe.android.core.utils.DurationProvider
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+internal class FakeDurationProvider(
+    private val duration: Duration = 1.seconds,
+) : DurationProvider {
+    private val calls: MutableList<Call> = mutableListOf()
```

Comment:
Can we use turbines instead? And add validation that all calls have been consumed?